runner/utils/driver/istio/istio_helpers.go

```
@@ -20,44 +20,94 @@ import (
 	"github.com/spaceuptech/space-cloud/runner/model"
 )
 
-func (i *Istio) prepareContainers(service *model.Service) []v1.Container {
+func (i *Istio) prepareContainers(service *model.Service) ([]v1.Container, []v1.Volume) {
 	// There will be n + 1 containers in the pod. Each task will have it's own container. Along with that,
 	// there will be a metric collection container as well which pushes metric data to the autoscaler.
 	// TODO: Add support for private repos
 	tasks := service.Tasks
 	containers := make([]v1.Container, len(tasks))
-	for i, task := range tasks {
-		// Prepare env variables
-		var envVars []v1.EnvVar
-		for k, v := range task.Env {
-			envVars = append(envVars, v1.EnvVar{Name: k, Value: v})
-		}
-
-		// Prepare ports to be exposed
-		ports := prepareContainerPorts(task.Ports)
-
-		// Prepare command and args
-		var cmd, args []string
-		if task.Docker.Cmd != nil {
-			cmd = task.Docker.Cmd[0:1]
-			if len(task.Docker.Cmd) > 1 {
-				args = task.Docker.Cmd[1:]
+	// Prepare Volume
+	volume := make([]v1.Volume, len(service.Secrets))
+	for j, task := range tasks {
+		for _, secretName := range service.Secrets {
+			// Prepare env variables
+			var envVars []v1.EnvVar
+			for k, v := range task.Env {
+				envVars = append(envVars, v1.EnvVar{Name: k, Value: v})
 			}
-		}
 
-		containers[i] = v1.Container{
-			Name: task.ID,
-			Env:  envVars,
+			// Prepare ports to be exposed
+			ports := prepareContainerPorts(task.Ports)
 
-			// Resource Related
-			Ports:     ports,
-			Resources: *generateResourceRequirements(&task.Resources),
+			// Prepare command and args
+			var cmd, args []string
+			if task.Docker.Cmd != nil {
+				cmd = task.Docker.Cmd[0:1]
+				if len(task.Docker.Cmd) > 1 {
+					args = task.Docker.Cmd[1:]
+				}
+			}
+			// get secrets
+			secrets, err := i.kube.CoreV1().Secrets(service.ProjectID).Get(secretName, metav1.GetOptions{})
+			if err != nil {
+				// error getting secrets!! return error!!
```

Maybe we should return an error here